server/plugin/api.go

```diff
@@ -817,6 +818,76 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
 				},
 			},
 		}
+	case constants.SubscriptionEventRunStageWaitingForApproval:
+		// TODO
+	case constants.SubscriptionEventReleaseDeploymentEventPending:
+		artifacts := ""
```

Comment:
I think everything inside this case should be moved to a separate function.

Reply:
for every case, we need to configure some fields so if we move them into a separate function we will need to do for all such fields also that function will not be used anywhere else except the needed case
let me know what you think

Reply:
You can merge it like this for now. We'll take a look at it later.